src/nvim/ex_docmd.c

```diff
@@ -6426,6 +6426,20 @@ ex_win_close(
   int need_hide;
   buf_T       *buf = win->w_buffer;
 
+  if (!tp) {
+    tp = curtab;
+  }
+
+  FOR_ALL_WINDOWS_IN_TAB(wp, tp)  {
```

Comment:
The problem is that this uses `wp->next`, which is invalid after `ex_win_close(forceit, wp, tp)`. We need to write out the loop explicitly and save `wp->next` before closing.

Reply:
Wouldn't `ex_win_close(forceit, wp, tp)` make `wp->next` `NULL`?